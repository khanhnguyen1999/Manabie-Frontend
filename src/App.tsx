import React from 'react';

import {BrowserRouter, Switch, Route} from 'react-router-dom';

import SignInPage from './features/Login/SignInPage';
import ToDoPage from './features/Todo/index';
import Error404 from './features/Error404View/Error404View';

import './App.css';

// routes
import AuthGuard from './guards/AuthGuard';
import GuestGuard from './guards/GuestGuard';

function App() {
  const accessToken = localStorage.getItem("token")
  return (
    <main className="App">
      <BrowserRouter>
        <Switch>
          <GuestGuard path="/login" isAuthenticated={accessToken ? true : false} component={SignInPage} />
          <AuthGuard path="/todo" isAuthenticated={accessToken ? true : false} component={ToDoPage} />
          <Route path="*" component={Error404} />
        </Switch>
      </BrowserRouter>
    </main>
  );
}

export default App;
