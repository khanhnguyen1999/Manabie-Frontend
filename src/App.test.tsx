import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ReactGA from 'react-ga';
import App from './App';
import renderer from 'react-test-renderer';

const mockStore = configureStore([]);

jest.mock('@material-ui/core', () => {
  const materialUI = jest.requireActual('@material-ui/core');
  return {
    ...materialUI,
    Menu: jest.fn(({ children, open }) => (open ? children : null)),
  };
});

jest.mock('@material-ui/core/Snackbar', () => ({
  __esModule: true,
  default: ({ children }: any) => <>{children}</>,
}));


jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useLayoutEffect: jest.requireActual('react').useEffect,
}));

describe('APP COMPONENT', () => {
  let store: any = null;
  let wrapper: any = null;

  beforeEach(() => {
    spyOn(ReactGA, 'initialize');
    store = mockStore({
      auth: { ACCESS_TOKEN: false },
    });
    wrapper = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    );
  });

  it('SHOULD MATCHED SNAPSHOT', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});