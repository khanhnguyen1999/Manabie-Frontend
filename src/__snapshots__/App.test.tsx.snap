// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP COMPONENT SHOULD MATCHED SNAPSHOT 1`] = `
<main
  className="App"
>
  <div
    className="MuiGrid-root makeStyles-loginContainer-2 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-content-xs-center"
  >
    <h1>
      Login
    </h1>
    <div
      className="MuiGrid-root makeStyles-content-5 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiFormControl-root makeStyles-form-9 MuiFormControl-fullWidth"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <div
            className="MuiInputBase-root makeStyles-input-10 MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input"
              disabled={false}
              name="userId"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Username"
              required={false}
              type="text"
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root makeStyles-form-9 MuiFormControl-fullWidth"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
            />
          </svg>
          <div
            className="MuiInputBase-root makeStyles-input-10 MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input"
              disabled={false}
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Password"
              required={false}
              type="password"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="makeStyles-root-11"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Login
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation6"
    role="alert"
  >
    <div
      className="MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message"
    >
      Login fail,please login again!
    </div>
  </div>
</main>
`;
